lib/IDE/CodeCompletion.cpp

```diff
@@ -5096,6 +5135,9 @@ void CodeCompletionCallbacksImpl::doneParsing() {
       CurDeclContext,
       CurDeclContext->getASTContext().SourceMgr.getCodeCompletionLoc());
 
+  // Add keywords even if type checking fails completely.
+  addKeywords(CompletionContext.getResultSink(), MaybeFuncBody);
```

Comment:
Moved after `typeCheckContextUntil()` because `addKeywords()` may call `getSelfProtocolDecl()` which can only be called after `bindExtensions()`.